opid/opid.go

```diff
@@ -0,0 +1,53 @@
+package opid
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
+
+// ContextKey is the type of keys used to store operation identifiers in contexts.
+type ContextKey int
+
+const (
+	// opidKey is the key used to store operation identifiers in context
+	opidKey ContextKey = iota
```

Comment:
I think the concept of the opid might warrant some additions to the README. Maybe a 'Usage' section can be helpful for other developers wanting to leverage this module. Not a blocker for me though.

Reply:
Yeah, there's more work to do on the documentation part. 
I'm also planning on adding more usage for all of this - WIP.

For now, concetrating on wiring it up with the backend as dependency, and then go back to complete the work here: pipeline, doc, etc.